File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        List<Integer> answer = new ArrayList<Integer>();
        for(int i=0;i<arr.length;i++){
            if(answer.size() == 0){
                answer.add(arr[i]);
            } else{
                if(answer.get(answer.size()-1) == arr[i]){
                    answer.remove(answer.size()-1);
                } else{
                    answer.add(arr[i]);
                }
            }
        }
        if(answer.size() == 0)
            answer.add(-1);
        return answer.stream().mapToInt(i->i).toArray();
    }
}

File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    answer = []
    for i in range(len(arr)):
        if not answer:
            answer.append(arr[i])
        else:
            if answer[-1] == arr[i]:
                answer.pop(len(answer)-1)
            else:
                answer.append(arr[i])
    if not answer:
        answer.append(-1)
    return answer